en/docs/develop-components/configure-endpoints-body.md

```diff
@@ -0,0 +1,124 @@
+
+Services and integrations are exposed to other services, integrations, or applications through endpoints. A service or an integration can expose multiple endpoints, each representing a unique entry point into the service. For example, a service may expose a REST API endpoint and a GraphQL endpoint, each providing different ways to interact with the service. Endpoints provide specific details for how the service or the integration can be consumed. For instance, the port number, protocol, and the schema such as open API specification (OAS) or GraphQL schema. By defining these details, endpoints make it possible for other services, integrations, and applications to discover and interact with the service in a standardized way.
+
+Choreo defines endpoints by combining port binding, protocol, endpoint name, network visibility, endpoint schema, and additional protocol-related fields. Following table describe each attribute of an endpoint.
+
+| Field | Description |
+| ----- | ----------- |
+| Name | A unique identifier for the endpoint within the service component. |
+| Port | The network port on which the endpoint is accessible. |
+| Type | The communication protocol used by the endpoint (e.g., HTTP, HTTPS, gRPC, etc.). |
+| Network Visibility | Determines the level of visibility of an endpoint. Possible values are: <ul><li>Project: Allows components within the same project to access the endpoint.</li><li>Organization: Allows any component within the same organization to access the endpoint but restricts access to components outside the organization.</li><li>Public: Allows any client to access the endpoint, regardless of location or organization.</li></ul> |
+| Schema | specifies the structure and format of the data exchanged through the endpoint. |
+| Context (HTTP and GraphQL only) | A context path that you add to the endpoint's URL for routing purposes. |
+
+## Configure endpoints
+
+The method of defining endpoints depends on the build preset. There are three build presets the supports defining endpoints: 
+
+- Ballerina
+- Dockerfile
+- WSO2 MI
+
+### Configure endpoints with the Dockerfile build-preset
+
+When you build a service component using the Dockerfile build-preset, you can configure the endpoint details with the endpoints.yaml configuration file. You must place this file inside the `.choreo` directory at the build context path and commit it to the source repository.
+
+See [Understanding the endpoints.yaml file](#understanding-the-endpointsyaml-file) to learn about the endpoints.yaml file.
```

Comment:
Shall we have `endpoints.yaml` in code format? Might have to apply this in all applicable places.

Reply:
done